src/olympia/addons/tests/test_views.py

```diff
@@ -1024,6 +1052,32 @@ def test_set_extra_data(self):
         assert data['support_url']['url'] == {'en-US': 'https://my.home.page/support/'}
         assert addon.support_url == 'https://my.home.page/support/'
 
+    def test_set_disabled(self):
+        response = self.client.patch(
+            self.url,
+            data={'is_disabled': True},
+        )
+        addon = Addon.objects.get()
+
+        assert response.status_code == 200, response.content
+        data = response.data
+        assert data['is_disabled'] is True
+        assert addon.is_disabled is True
+        assert addon.disabled_by_user is True  # sets the user property
+
+        # Confirm that a STATUS_DISABLED can't be overriden
+        addon.update(status=amo.STATUS_DISABLED)
+        response = self.client.patch(
+            self.url,
+            data={'is_disabled': False},
```

Comment:
Should this actually raise an exception? It seems like an invalid API call.

Reply:
I was thinking because they are trying to un-disable a disabled add-on and we don't allow that. So it seems like maybe we should return an error rather then just not allowing it to happen?

Reply:
There's a wider lack of enforcement of the restriction around disabled add-ons, so I filed #18330 to cover everything.  (The test is really more about checking a write to that property is writing to the correct model field; once #18330 is done we'll likely have to drop this test because it won't be possible anyway)